controllers/prep_handlers.go

```diff
@@ -68,8 +69,56 @@ func (r *ImageBasedUpgradeReconciler) getSeedImage(
 		return fmt.Errorf("failed to pull image: %w", err)
 	}
 
+	r.Log.Info("Checking seed image compatibility")
+	if err := r.checkSeedImageCompatibility(ctx, pullSecretFilename, ibu.Spec.SeedImageRef.Image); err != nil {
+		return fmt.Errorf("checking seed image compatibility: %w", err)
+	}
+
 	return nil
+}
+
+// checkSeedImageCompatibility checks if the seed image is compatible with the
+// current version of the lifecycle-agent by inspecting the OCI image's labels
+// and checking if the specified format version equals the hard-coded one that
+// this version of the lifecycle agent expects. That format version is set by
+// the imager during the image build process, and is only manually bumped by
+// developers when the image format changes in a way that is incompatible with
+// previous versions of the lifecycle-agent.
+func (r *ImageBasedUpgradeReconciler) checkSeedImageCompatibility(ctx context.Context, pullSecretFilename string, seedImageRef string) error {
+	inspectArgs := []string{
+		"inspect",
+		"--format", "json",
+		"--authfile", pullSecretFilename,
+		seedImageRef,
+	}
+
+	var inspect struct {
+		Labels map[string]string `json:"Labels"`
+	}
+	if inspectRaw, err := r.Executor.Execute("podman", inspectArgs...); err != nil || inspectRaw == "" {
```

Comment:
One advantage to your method of adding the compatibility version in a label means we could potentially inspect the image manifest before pulling it, such as with ```skopeo inspect```. Given that the images are more than 1G, that could be very beneficial.

Reply:
Yep. Will leave that for a future PR

Reply:
Created #201